modules/paapiForGpt.js

```diff
@@ -3,26 +3,39 @@
  */
 import {getHook, submodule} from '../src/hook.js';
 import {deepAccess, logInfo, logWarn, sizeTupleToSizeString} from '../src/utils.js';
-import {getGptSlotForAdUnitCode} from '../libraries/gptUtils/gptUtils.js';
 import {config} from '../src/config.js';
 import {getGlobal} from '../src/prebidGlobal.js';
 
 import {keyCompare} from '../src/utils/reducers.js';
+import {getGPTSlotsForAdUnits, targeting} from '../src/targeting.js';
+
 const MODULE = 'paapiForGpt';
 
 let getPAAPIConfig;
 
-let autoconfig = false;
-
 config.getConfig('paapi', (cfg) => {
-  autoconfig = deepAccess(cfg, 'paapi.gpt.autoconfig', false);
+  if (deepAccess(cfg, 'paapi.gpt.configWithTargeting', false)) {
```

Comment:
yes. are you saying that you'd prefer the default to be `true`? cc @patmmccann

Reply:
As long as the publisher has to manually call `pbjs.setPAAPIConfigForGPT()` and fails to do so for subsequent refresh calls, the caching issue described in [#11242](https://github.com/prebid/Prebid.js/issues/11242) can still happen. Setting it to true by default (like before) would help, but if disabled and not properly configured, the caching issue will also happen again.

Reply:
Are you proposing a second check in set targeting that checks for no stale configs?

Reply:
Yes that would work or we can always run `setPAAPIConfigForGPT()` as part of `setTargetingForGPTAsync`. This would ensure the latest auction configs are configured regardless of `configWithTargeting` mode.

Reply:
The latter is known to cause a variety of problems and won't work with #11730 

@dgirardi any thoughts on the former? Targeting can detect and clear stale config if one is using paapi?

Reply:
IMO:
 - https://github.com/prebid/Prebid.js/issues/11242 is not caused by people having incorrect setup, and whatever we do here will not fix it
 - throwing away unused config on targeting is worse than automatically using it, harder to understand as a pub, it's an obscure fix to an obscure problem that I don't think will not even work as a fix
 - what's the problem with setting `configWithTargeting: true`? it's on a major version release that will include many other breaking changes, I don't think this will be the thing preventing people from updating.

Reply:
I'm inclined to agree, this appears to solve making things easier for publishers, and ix can submit further non-breaking guardrails at a later date

Reply:
Ix can also prevent stale config from being reused by hard coding dummy config into their adapter and replacing with the response config, no?

Reply:
That won't work. Since if `pbjs.setPAAPIConfigForGPT()` isn't called as part of the refresh call the new auction config isn't associated with the GPT slot. This issue isn't specific to the IX bidder.

Reply:
Updated to reset slots on targeting regardless of `configWithTargeting`.

Reply:
@lksharma are we clear to merge?

Reply:
@mkendall07 -- can you dig in further?   I do think we need to understand the interplay between auto-config and this new behaviour.